### PR TITLE
Fusion data flow

### PR DESCRIPTION
**Pull request by victoryforphil**
## Summary
- Moved to new JSON Library
- Fixed / Enabled the collision dropdown
- Fixed JSON exporting
- general stability improvements.

## Table
|    DOD    | Done |    Reviewer    | Issue | Notes |
|-----------|------|----------------|-------|-------|
| Code      | ❌   | Nicolas Espinoza|  n/a|       |
| Unit Test | N/A  | N/A            |       |       |
| Merged    | ❌  |                |       |       |

### CODE REVIEW

File: exporters/FusionRobotExporter/Source/AddIn/CustomHandlers.cpp [hunk 1]
```diff
@@ -70,7 +70,7 @@ void ShowPaletteCommandExecuteHandler::notify(const Ptr<CommandEventArgs>& event
 		eui->openDriveWeightPalette();
 	else if (id == SynthesisAddIn::BTN_EDIT_JOINTS)
 		eui->openJointEditorPalette();
-	//else if (id == SynthesisAddIn::BTN_DOF) {
+	//else if (id == Synthe sisAddIn::BTN_DOF) {
```
Comment (wangl5) [inline exporters/FusionRobotExporter/Source/AddIn/CustomHandlers.cpp]:
```suggestion
	//else if (id == SynthesisAddIn::BTN_DOF) {
```

File: exporters/FusionRobotExporter/Source/Data/BXDJ/ConfigData.cpp [hunk 2]
```diff
@@ -178,164 +178,126 @@ void ConfigData::filterJoints(std::vector<core::Ptr<fusion::AsBuiltJoint>> filte
...
 		// Joint Info
-		jointJSON.AddMember("id", rapidjson::Value(jointID.c_str(), jointID.length(), allocator), allocator);
-		jointJSON.AddMember("name", rapidjson::Value(jointConfig.name.c_str(), jointConfig.name.length(), allocator), allocator);
-		jointJSON.AddMember("asBuilt", jointConfig.asBuilt, allocator);
-		jointJSON.AddMember("type", rapidjson::Value((int)jointConfig.motion), allocator);
-
-
-
+		jointJSON["id"] = jointID;
+		jointJSON["name"] = jointConfig.name;
```
Comment (wangl5) [inline exporters/FusionRobotExporter/Source/Data/BXDJ/ConfigData.cpp]: Could you add the joint weight to the JSON parsing?

File: exporters/FusionRobotExporter/Source/Data/BXDJ/Driver.cpp [hunk 3]
```diff
@@ -127,39 +128,46 @@ nlohmann::json BXDJ::Driver::GetExportJson()
...
+		}
+
+		if (driverJSON["portOne"].is_number()) {
+			portOne = driverJSON["portOne"].get<int>();
+		}
+
+		if (driverJSON["portTwo"].is_number()) {
+			portTwo = driverJSON["portTwo"].get<int>();
+		}
+
```
Comment (wangl5) [inline exporters/FusionRobotExporter/Source/Data/BXDJ/Driver.cpp]: the `inputGear` and `outputGear` fields should also be assigned here.

File: exporters/FusionRobotExporter/Source/Exporter.cpp [hunk 4]
```diff
@@ -178,7 +178,7 @@ void Exporter::exportMeshes(BXDJ::ConfigData config, Ptr<FusionDocument> documen
 		baseJson["SoftwareExportedWith"] = "FUSION_360";
```
Comment (wangl5) [inline exporters/FusionRobotExporter/Source/Exporter.cpp]: Where does the drivetrain weight get exported?

File: exporters/FusionRobotExporter/Source/Data/BXDJ/ConfigData.cpp (hunk 2 above)
Comment (victoryforphil) [reply exporters/FusionRobotExporter/Source/Data/BXDJ/ConfigData.cpp]: Ill do that in a new branch where we do per joint weights. Not fitting in this one.

File: exporters/FusionRobotExporter/Source/Data/BXDJ/Driver.cpp (hunk 3 above)
Comment (victoryforphil) [reply exporters/FusionRobotExporter/Source/Data/BXDJ/Driver.cpp]: This is handled in the wheel json

File: exporters/FusionRobotExporter/Source/Exporter.cpp (hunk 4 above)
Comment (victoryforphil) [reply exporters/FusionRobotExporter/Source/Exporter.cpp]: No where. Not in current JSON spec/engine. Looking at putting it as a joint weight for the parent node

File: exporters/FusionRobotExporter/Source/Data/BXDJ/Driver.cpp (hunk 3 above)
Comment (wangl5) [reply exporters/FusionRobotExporter/Source/Data/BXDJ/Driver.cpp]: Aren't all joint drivers supposed to have an `inputGear` and `outputGear`?

**Comment by wangl5** [comment]
https://github.com/Autodesk/synthesis/pull/419#discussion_r312197411

File: exporters/FusionRobotExporter/Source/Data/BXDJ/Driver.cpp (hunk 3 above)
Comment (victoryforphil) [reply exporters/FusionRobotExporter/Source/Data/BXDJ/Driver.cpp]: Correct. Fixed.

**Pull request merged**